File: src/main/java/org/fabiomsr/rxjava/_00_creators/_00_Just.java
package org.fabiomsr.rxjava._00_creators;

import rx.Observable;

/**
 * Created by fabiomsr on 29/10/16.
 */
public class _00_Just {

    private static void java() {
        System.out.println(1);
        System.out.println(2);
        System.out.println(3);
    }

    private static void rxJava() {
        Observable.just(1, 2, 3)
                  .forEach(number -> System.out.println(number));
        // or .forEach(System.out::println)
        // or .subscribe(System.out::println)
    }

    public static void main(String[] args) {
        System.out.println("Result: Java version");
        java();
        System.out.println("Result: RxJava version");
        rxJava();
    }

}

File: src/main/kotlin/org/fabiomsr/rxkotlin/_00_creators/_00_Just.kt
package org.fabiomsr.rxkotlin._00_creators

import rx.Observable

/**
 * Created by fabiomsr on 29/10/16.
 */

private fun kotlin() {
    println(1)
    println(2)
    println(3)
}

private fun rxKotlin() {
    Observable.just(1, 2, 3)
              .forEach { number -> println(number) }
    // or .forEach { println(it) }
    // or .forEach(::println)
    // or .subscribe(::println)
}

fun main(args : Array<String>){
    println ("Result: Kotlin version")
    kotlin()
    println ("Result: RxKotlin version")
    rxKotlin()
}